hscontrol/grpcv1.go

```diff
@@ -652,6 +654,64 @@ func (api headscaleV1APIServer) DeleteApiKey(
 	return &v1.DeleteApiKeyResponse{}, nil
 }
 
+// GetACL returns the current ACL.
+func (api headscaleV1APIServer) GetACL(
+	_ context.Context,
+	_ *v1.GetACLRequest,
+) (*v1.GetACLResponse, error) {
+	acl, err := api.h.db.GetACL()
+	if err != nil {
+		if db.IsNotFoundError(err) {
+			return nil, types.ErrACLPolicyNotFound
+		}
+		return nil, err
+	}
+
+	return &v1.GetACLResponse{
+		Policy: acl.Proto().GetPolicy(),
+	}, nil
+}
+
+// SetACL inserts or updates the ACL.
+func (api headscaleV1APIServer) SetACL(
+	_ context.Context,
+	request *v1.SetACLRequest,
+) (*v1.SetACLResponse, error) {
+	if api.h.cfg.ACL.PolicyMode != types.ACLPolicyModeDB {
+		return nil, types.ErrACLPolicyUpdateIsDisabled
+	}
+
+	polBytes, err := request.GetPolicy().MarshalJSON()
```

Comment:
Would this fail if the input is HuJSON (default Tailscale ACL format)? `policy.LoadACLPolicyFromBytes` will use the hujson parser by default when the format is not `yaml`.

Reply:
Thanks for bringing this up. I will test and verify it.